lib/fs.js

```diff
@@ -1384,6 +1384,33 @@ function mkdirSync(path, options) {
   }
 }
 
+function readdirSyncRecursive(origPath, options) {
+  function _readdirSyncRecursive(path) {
+    path = getValidatedPath(path);
```

Comment:
I don't know if we need to call this recursively - just on `origPath`. Out of everything that `getValidatedPath()` does, the only thing we might need to do recursively is call [`nullCheck()`](https://github.com/nodejs/node/blob/01323d50c4b24cf730a651d06ba20633905ecbed/lib/internal/fs/utils.js#L678).